File: sso-auth/src/main/java/com/study/infosecurity/ssoByRoles/repository/UserRepository.java
package com.study.infosecurity.ssoByRoles.repository;

import com.study.infosecurity.ssoByRoles.model.dto.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository("userRepository")
public interface UserRepository extends JpaRepository<User, Long> {
    List<User> findByUsername(String username);
}

File: sso-model/src/main/kotlin/com/study/infosecurity/ssoByRoles/model/repository/UserRepository.kt
package com.study.infosecurity.ssoByRoles.model.repository

import com.study.infosecurity.ssoByRoles.model.dto.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository("userRepository")
interface UserRepository : JpaRepository<User, Long> {
    fun findByUsername(username: String): List<User>
}